### DISCUSSION TITLE
Generic type plotting

### DISCUSSION THREAD

**Discussion by MattOslin** [foxglove/community#91] [Q&A]
Hello, I am trying to plot a generic message (for purposes of this discussion it can be a list of key value pairs, where the keys are the trace names and the values are the traces). It appears that in the typescript interface I have to explicitly lay out the return type, and generic fields like Map don't appear to be supported. Furthermore, for each trace I have to manually add it to the plot: the plotting tool cannot seem to unpack more than one trace at a time. Is there any planned feature work to support plotting more generic types like maps from trace names to traces?

**Comment by defunctzombie** [comment]
Can you provide an example of a message schema and data that you are trying to plot? Are you trying to use user-scripts (based on your question about the typescript interface) or something else? I'm having some trouble following the specific question. If you can elaborate with some examples we might be able to point you in the right direction or better understand the feature ask.
